src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -281,6 +283,8 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 		column->varno = OUTER_VAR;
 	}
 
+	hasDistinctAggregate = HasDistinctAggregate(masterQuery);
```

Comment:
We can move declaration and assignment of `hasDistinctAggregate ` to inside `if (groupColumnCount > 0)` where it is only used.

Reply:
sounds good